### PR TITLE
Release notes for Release 3.0.1

### PR DESCRIPTION
**Pull request by awholegunch**
Subject to review and change.

Staged at: [Release Notes](http://docs-staging.cask.co/cdap/3.0.1-feature-r3.0.1_release_notes/en/reference-manual/release-notes.html#release-3-0-1)

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -24,6 +24,271 @@ Cask Data Application Platform Release Notes
...
+
+
+Improvements
+------------
+
+**Tools**
+
+- Added a tool (`HBaseQueueDebugger 
+  <https://github.com/caskdata/cdap/blob/release/3.0/cdap-master/src/main/java/co/cask/cdap/data/tools/HBaseQueueDebugger.java>`__)
+  that counts consumed and unconsumed entries in an HBase queue
```
Comment (jwang47) [inline cdap-docs/reference-manual/source/release-notes.rst]: Probably "flowlet queue" instead of "HBase queue"

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Hmm. Is the tool mis-named then? "FlowletQueueDebugger.java"?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -24,6 +24,271 @@ Cask Data Application Platform Release Notes
...
 
 
+`Release 3.0.1 <http://docs.cask.co/cdap/3.0.1/index.html>`__
+=============================================================
+
+
+New Features
+------------
+
+- Added the CDAP Application Templates (``cdap-app-templates``) to the `CDAP Javadocs
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: We can remove this.

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (jwang47) [inline cdap-docs/reference-manual/source/release-notes.rst]: No, the name means "debugger for queues implemented using HBase tables". We have lots of classes like HBaseQueue* and LevelDBQueue*.

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Pull request merged**